pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -0,0 +1,326 @@
+	//regardless if it is indexed/present again in resource or column attribute
+	_, isTopLevelCol := constants.LogsTopLevelColumnsV3[key.Key]
+	if !isTopLevelCol && !key.IsColumn {
+		// check if we need to enrich metadata
+		if key.Type == "" || key.DataType == "" {
+			// check if the field is present in the fields map
+			if field, ok := fields[key.Key]; ok {
+				if field.IsColumn {
+					return field.Key, nil
+				}
+				key.Type = field.Type
+				key.DataType = field.DataType
+			} else {
+				return "", fmt.Errorf("field not found to enrich metadata")
+			}
+		}
+
+		columnType := key.Type
+		if columnType == v3.AttributeKeyTypeTag {
+			columnType = "attributes"
+		} else {
+			columnType = "resources"
+		}
+
+		columnDataType := "string"
+		if key.DataType == v3.AttributeKeyDataTypeFloat64 {
+			columnDataType = "float64"
+		}
+		if key.DataType == v3.AttributeKeyDataTypeInt64 {
+			columnDataType = "int64"
+		}
+		clickhouseColumn = fmt.Sprintf("%s_%s_value[indexOf(%s_%s_key, '%s')]", columnType, columnDataType, columnType, columnDataType, key.Key)
+	}
+	return clickhouseColumn, nil
+}
+
+// getSelectLabels returns the select labels for the query based on groupBy and aggregateOperator
+func getSelectLabels(aggregatorOperator v3.AggregateOperator, groupBy []v3.AttributeKey, fields map[string]v3.AttributeKey) (string, error) {
+	var selectLabels string
+	if aggregatorOperator == v3.AggregateOperatorNoOp || aggregatorOperator == v3.AggregateOperatorRate {
+		selectLabels = ""
+	} else {
+		for _, tag := range groupBy {
+			columnName, err := getClickhouseColumnName(tag, fields)
+			if err != nil {
+				return "", err
+			}
+			selectLabels += fmt.Sprintf(", %s as %s", columnName, tag.Key)
+		}
+	}
+	return selectLabels, nil
+}
+
+func buildLogsTimeSeriesFilterQuery(fs *v3.FilterSet, fields map[string]v3.AttributeKey) (string, error) {
+	var conditions []string
+
+	if fs != nil && len(fs.Items) != 0 {
+		for _, item := range fs.Items {
+			toFormat := item.Value
+			op := strings.ToLower(strings.TrimSpace(item.Operator))
+
+			// generate the key
+			columnName, err := getClickhouseColumnName(item.Key, fields)
+			if err != nil {
+				return "", err
+			}
+			fmtVal := utils.ClickHouseFormattedValue(toFormat)
+
+			if operator, ok := logsOperatorMappingV3[op]; ok {
+				conditions = append(conditions, fmt.Sprintf("%s %s %s", columnName, operator, fmtVal))
+			} else {
+				return "", fmt.Errorf("unsupported operation")
+			}
+		}
+	}
+	queryString := strings.Join(conditions, " AND ")
+
+	if len(queryString) > 0 {
+		queryString = " AND " + queryString
+	}
+	return queryString, nil
+}
+
+func buildLogsQuery(start, end, step int64, mq *v3.BuilderQuery, tableName string, fields map[string]v3.AttributeKey) (string, error) {
+
+	filterSubQuery, err := buildLogsTimeSeriesFilterQuery(mq.Filters, fields)
+	if err != nil {
+		return "", err
+	}
+
+	samplesTableTimeFilter := fmt.Sprintf("(timestamp >= %d AND timestamp <= %d)", start, end)
```

Comment:
```suggestion
	timeFilter := fmt.Sprintf("(timestamp >= %d AND timestamp <= %d)", start, end)
```

Check the precision

Reply:
Ahh so in my payload I was sending Epochnano that's why missed this. Keeping it epoch milliseconds for all.

Reply:
Added a function to which will convert it to nano if the input is in milli/micro.